corehq/apps/reports/forms.py

```diff
@@ -313,6 +326,21 @@ def __init__(self, data, *args, **kwargs):
             )
         )
 
+        if self.add_allowed_tableau_groups_field:
```

Comment:
Can the ` if user_syncing_config['user_syncing_enabled'] and user_syncing_config.get('server_reachable'):` conditional check be moved down here and not need to check `  if self.add_allowed_tableau_groups_field:` here?